### PR TITLE
Feature/logout

### PR DESCRIPTION
**Pull request by Nazaqat**

### CODE REVIEW

File: src/client/components/Logout/Logout.component.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import Popup from 'reactjs-popup';
+import './Logout.style.css';
+
+const userName = 'Some UserName';
```
Comment (zkwsk) [inline src/client/components/Logout/Logout.component.js]: No hardcoded values in your component. Test data like that belongs in Storybook.

File: src/client/components/Logout/Logout.component.js [hunk 2]
```diff
@@ -2,11 +2,19 @@ import React from 'react';
 import Popup from 'reactjs-popup';
 import './Logout.style.css';
 
+const userName = 'Some UserName';
+
 export default function Logout() {
+  function logout(event) {
+    event.preventDefault(); // prevent page transition
+    fetch('/logout', { method: 'POST' }).then(
```
[the remark this answers is not in the document]
Comment (zkwsk) [reply src/client/components/Logout/Logout.component.js]: Please remove the "logic" to log out. We don't know how that is going to work at the moment.

Instead take a logout function as a prop and trigger that when user clicks on the logout button, so we can control this logic from the parent. This is a presentational component, not a container component, so it should have no logic itself.

File: src/client/components/Logout/Logout.component.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+              Logout
+            </button>
+            <br />
+            <button
+              type="submit"
+              className="ConfirmCancel"
+              onClick={() => {
+                close();
+              }}
+            >
```
Comment (zkwsk) [inline src/client/components/Logout/Logout.component.js]: Look into re-using the button component in @cecastosic 's PR.

File: src/client/components/Logout/Logout.story.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
+import React from 'react';
+import { storiesOf } from '@storybook/react';
+import Logout from './Logout.component';
+
+storiesOf('Logouts', module).add('Logout Popup', () => <Logout />);
```
Comment (zkwsk) [inline src/client/components/Logout/Logout.story.js]: Please update your storybook syntax. Check the guide in the readme file.

File: src/client/components/Logout/Logout.style.css [hunk 5]
```diff
@@ -0,0 +1,114 @@
+/*Log button on front end styling */
+.logbutton {
```
Comment (zkwsk) [inline src/client/components/Logout/Logout.style.css]: A "log" is somewhere you can list events, such as in a log book or something like that, so I think the name `logbutton` is confusing. Use `logout-button` instead. Or maybe you don't need any class at all if you re-use the button component as discussed above.

File: src/server/knexfile.ts [hunk 6]
```diff
@@ -0,0 +1,44 @@
+// Update with your config settings.
+
+module.exports = {
```
Comment (zkwsk) [inline src/server/knexfile.ts]: Creating a typescript file?? What's going on here. Please delete this file. I think you might have run some command in knex that initialized knex with a new config file. Make sure to read your PR through before creating it so you can catch files that might be created by accident.

File: src/client/components/Logout/Logout.component.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (Nazaqat) [reply src/client/components/Logout/Logout.component.js]: Code reviewed

File: src/client/components/Logout/Logout.component.js [hunk 7]
```diff
@@ -2,11 +2,19 @@ import React from 'react';
...
+const userName = 'Some UserName';
+
 export default function Logout() {
+  function logout(event) {
+    event.preventDefault(); // prevent page transition
+    fetch('/logout', { method: 'POST' }).then(
+      () => window.location.reload(), // stay at the same url
+    );
+  }
   return (
```
[the remark this answers is not in the document]
Comment (Nazaqat) [reply src/client/components/Logout/Logout.component.js]: Done.

File: src/client/components/Logout/Logout.component.js (hunk 1 above)
Comment (Nazaqat) [reply src/client/components/Logout/Logout.component.js]: Done

File: src/client/components/Logout/Logout.story.js (hunk 4 above)
Comment (Nazaqat) [reply src/client/components/Logout/Logout.story.js]: ok

File: src/server/knexfile.ts (hunk 6 above)
Comment (Nazaqat) [reply src/server/knexfile.ts]: Deleted

File: src/client/components/Logout/Logout.component.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (Nazaqat) [reply src/client/components/Logout/Logout.component.js]: Got it

File: src/client/components/Logout/Logout.component.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (Nazaqat) [reply src/client/components/Logout/Logout.component.js]: Removed

**Comment by Nazaqat** [comment]
updated version will be uploaded soon

**Pull request closed without merging**